src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -1416,6 +1416,114 @@ def _get_iot_central_app_by_name(client, app_name):
     return target_app
 
 
+def iot_central_list_private_link_resource(client, app_name=None, connection_id=None, resource_group_name=None, source_type=None):
+    if app_name and resource_group_name and source_type:
+        if source_type != 'Microsoft.IoTCentral/iotApps':
+            raise CLIError(
+                "Type must be Microsoft.IoTCentral/iotApps")
+    elif connection_id:
+        id_list = connection_id.split('/')
+        resource_group_name = id_list[id_list.index('resourceGroups') + 1]
+        app_name = id_list[id_list.index('iotApps') + 1]
+    else:
+        raise CLIError(
+            "Must provide id or application name, resource group, and type.")
```

Comment:
Could you please use the specific error type `RequiredArgumentMissingError` instead of `CLIError`?